src/Imath/half.h

```diff
@@ -9,188 +9,209 @@
-//	exactly.  If a float is not representable as a half, then the
-//	float value is rounded to the nearest representable half.  If a
-//	float value is exactly in the middle between the two closest
-//	representable half values, then the float value is rounded to
-//	the closest half whose least significant bit is zero.
-//
-//	Overflows during float-to-half conversions cause arithmetic
-//	exceptions.  An overflow occurs when the float value to be
-//	converted is too large to be represented as a half, or if the
-//	float value is an infinity or a NAN.
-//
-//	The implementation of type half makes the following assumptions
-//	about the implementation of the built-in C++ types:
-//
-//	    float is an IEEE 754 single-precision number
-//	    sizeof (float) == 4
-//	    sizeof (unsigned int) == sizeof (float)
-//	    alignof (unsigned int) == alignof (float)
-//	    sizeof (unsigned short) == 2
-//
-//---------------------------------------------------------------------------
-
 #ifndef _HALF_H_
 #define _HALF_H_
 
-/// @cond Doxygen_Suppress
-
 #include "ImathNamespace.h"
 #include "ImathExport.h"
 #include "ImathPlatform.h"
 #include <iostream>
 
 IMATH_INTERNAL_NAMESPACE_HEADER_ENTER
 
+///
+///
+/// class half -- 16-bit floating point number
+///
+/// Type half can represent positive and negative numbers whose
+/// magnitude is between roughly 6.1e-5 and 6.5e+4 with a relative
+/// error of 9.8e-4; numbers smaller than 6.1e-5 can be represented
+/// with an absolute error of 6.0e-8.  All integers from -2048 to
+/// +2048 can be represented exactly.
+///
+/// Type half behaves (almost) like the built-in C++ floating point
+/// types.  In arithmetic expressions, half, float and double can be
+/// mixed freely.  Here are a few examples:
+///
+///     half a (3.5);
+///     float b (a + sqrt (a));
+///     a += b;
+///     b += a;
+///     b = a + 7;
+///
+/// Conversions from half to float are lossless; all half numbers
+/// are exactly representable as floats.
+///
+/// Conversions from float to half may not preserve a float's value
+/// exactly.  If a float is not representable as a half, then the
+/// float value is rounded to the nearest representable half.  If a
+/// float value is exactly in the middle between the two closest
+/// representable half values, then the float value is rounded to
+/// the closest half whose least significant bit is zero.
+///
+/// Overflows during float-to-half conversions cause arithmetic
+/// exceptions.  An overflow occurs when the float value to be
+/// converted is too large to be represented as a half, or if the
+/// float value is an infinity or a NAN.
+///
+/// The implementation of type half makes the following assumptions
+/// about the implementation of the built-in C++ types:
+///
+/// * float is an IEEE 754 single-precision number
+/// * sizeof (float) == 4
+/// * sizeof (unsigned int) == sizeof (float)
+/// * alignof (unsigned int) == alignof (float)
+/// * sizeof (unsigned short) == 2
```

Comment:
Note for someday: These should be verified by static_assert. And all the int types mentioned should be uintN_t, not plain int or short whose size is not nailed down by the C++ standard.

Reply:
#118